sycl/doc/SharedLibraries.md

```diff
@@ -0,0 +1,276 @@
+# Shared DPC++ libraries
+
+This document describes purpose and design of Shared DPC++ libraries feature.
```

Comment:
I think shared dpc++ libraries are already supported per se - this is a .dll with device binary image containing kernels.

The key new functionality is dynamic linkage of device code. Moreover, I think "shared device library" is not really accurate term as the dynamically linked device code is not shared between device images, but duplicated. So I would suggest to make that distinction throughout the document.

Reply:
> Moreover, I think "shared device library" is not really accurate term as the dynamically linked device code is not shared between device images, but duplicated.

From "library" point of view, it is shared. The fact that within it some source code might be duplicated doesn't make it not shared

Reply:
I agree that shared libraries already supported in some form. We can rename it to something like "automatic dynamic linkage of device code".

Reply:
> > Moreover, I think "shared device library" is not really accurate term as the dynamically linked device code is not shared between device images, but duplicated.
> 
> From "library" point of view, it is shared. The fact that within it some source code might be duplicated doesn't make it not shared

It actually does. Otherwise we could call static libraries shared, but we don't. I do believe for people with traditional C/C++ background and not familiar with SYCL offload peculiarities "shared" will sound confusing.

Reply:
Renamed the feature.